roles/kafka_controller/tasks/health_check.yml

```diff
@@ -1,26 +1,62 @@
 ---
 # health check for kafka controller
-- name: Check Kafka Metadata Quorum
+- name: Check Kafka Metadata Quorum using bootstrap controller
   shell: |
     {{ binary_base_path }}/bin/kafka-metadata-quorum --bootstrap-controller {{inventory_hostname}}:{{kafka_controller_port}} \
       --command-config {{kafka_controller.client_config_file}} describe --replication
   environment:
     KAFKA_OPTS: "-Xlog:all=error -XX:+IgnoreUnrecognizedVMOptions {% if kerberos_client_config_file_dest != '/etc/krb5.conf' %}-Djava.security.krb5.conf={{kerberos_client_config_file_dest}}{% endif %}"
+  ignore_errors: true
+  changed_when: false
+  check_mode: false
+  register: greenfield
+
+- name: Set Hostnames and Ports
+  set_fact:
+    broker_hostname_hc: "{{ hostvars[groups['kafka_broker'][0]] | confluent.platform.resolve_hostname }}"
+    broker_port_hc: "{{kafka_broker_listeners[kafka_broker_inter_broker_listener_name]['port']}}"
+    controller_hostname_hc: "{{inventory_hostname}}"
+    controller_port_hc: "{{kafka_controller_port}}"
+
+- name: Set Server/Controller Host and Port values
+  set_fact:
+    bootstrap_server_or_controller: "{% if greenfield.rc == 0 %}controller{% elif greenfield.rc != 0 and confluent_server_enabled|bool %}server{% endif %}"
+    server_hostname: "{% if greenfield.rc == 0 %}{{controller_hostname_hc}}{% elif greenfield.rc != 0 and confluent_server_enabled|bool %}{{broker_hostname_hc}}{% endif %}"
+    server_port: "{% if greenfield.rc == 0 %}{{controller_port_hc}}{% elif greenfield.rc != 0 and confluent_server_enabled|bool %}{{broker_port_hc}}{% endif %}"
+
+- name: Add confluent.use.controller.listener config to Client Properties
```

Comment:
forgot to mention, this config won't be required anymore. 7.7 uses bootstra-controller, and in upgrade we are using broker server which doesn't need this config. Most probably the support has also been dropped

Reply:
Oh yes. Will remove this and its counterpart task to remove this line